src/Ombi.Store/Repository/Requests/MovieRequestRepository.cs

```diff
@@ -46,12 +46,18 @@ public MovieRequests GetRequest(int theMovieDbId)
                 .FirstOrDefault();
         }
 
-        public IQueryable<MovieRequests> GetWithUser()
+        public IQueryable<MovieRequests> GetWithUser(bool anonimize)
         {
-            return Db.MovieRequests
-                .Include(x => x.RequestedUser)
-                .ThenInclude(x => x.NotificationUserIds)
-                .AsQueryable();
+            if (!anonimize)
+            {
+                return Db.MovieRequests
+                    .Include(x => x.RequestedUser)
+                    .ThenInclude(x => x.NotificationUserIds)
+                    .AsQueryable();
+            } else
+            {
+                return Db.MovieRequests.AsQueryable(); //This still populates the RequestedUser for the logged in user (or so it seems...)
```

Comment:
Had a quick look at the docs, looks like this is because that EF Core has already potentially cached the data from another execution and it's using a shared database context.

It might be better to revert the changes to this `GetWithUser` method and instead create a new method that returns 

```
 return Db.MovieRequests.AsNoTracking().AsQueryable();
 ```

That will then disable the caching for that entity, but it also means that we cannot perform any updates to the entity without re-attaching to the context, so that's why I wouldn't do it as part o the `GetWithUser` method.


Instead maybe create a common method in the `MovieRequestEngine`, something like `GetRequests(bool anonymize)` and then either call `GetWithUser` or your new method depending on the value of `anonymize`

What do you think?

Reply:
Seems good, alternative option would be to pass a default parameter in `anonimize`, preferably false, as that would result in the same logic but without the split in the method but that's codestyling preference i guess?

Reply:
Yeah that works for me too.